src/compiler/checker.ts

```diff
@@ -12849,13 +12849,15 @@ namespace ts {
                         }
                     }
                     else if (target.flags & TypeFlags.Intersection) {
-                        isIntersectionConstituent = true; // set here to affect the following trio of checks

```

Comment:
So this affected the `recursiveTypeRelatedTo` call after this `else if` block (if the contents of the block did not cause a return) before (the third check of the `trio` referenced in the comment), but now nothing does. Was the change intentional?

Reply:
I *think* so (the intervening vacation makes things a bit fuzzy). I think I decided it could be dispensed with entirely because isIntersectionConstituent is now set in typeRelatedToEachType (line 13152) instead of here. But it may be there is a difference and that our tests just don't catch it.

In any case, after leaving this PR for a while, I kind of like the threaded-parameter approach. I'm going to measure performance on both commits to make sure that there are no weird limits on parameter count in the VM and go back to explicit parameters if there's no perf difference.